packages/react/src/Dialog/Dialog.tsx

```diff
@@ -27,7 +29,7 @@ export const Dialog = forwardRef(
       <form method="dialog" className="ams-dialog__form">
         <header className="ams-dialog__header">
           <Heading size="level-4">{heading}</Heading>
-          <IconButton formNoValidate label={closeButtonLabel} size="level-4" />
+          <IconButton label={closeButtonLabel} size="level-4" type="button" onClick={closeDialog} />
```

Comment:
Why do we need `type="button"` again?

Reply:
A `button` element has `type="submit"` by default. There is a discrepancy there though. We change this default to `type="button"` for our Button component. Do we want to change this for IconButton as well? Makes sense to keep those in sync imo

Reply:
Yes, I’ll create a separate task for that.